### PR TITLE
Refactoring related to new provider types integration

### PR DESCRIPTION
**Pull request by EugeneButusov**

### CODE REVIEW

File: SoomlaiOSProfile/ConfirmationDialog.h [hunk 1]
```diff
@@ -4,7 +4,7 @@
 //
 
 #import <Foundation/Foundation.h>
-
+#import <UIKit/UIKit.h>
```
Comment (borisspektor) [inline SoomlaiOSProfile/ConfirmationDialog.h:7]: @EugeneButusov why is this needed now/why wasn't this here before?

Comment (EugeneButusov) [inline SoomlaiOSProfile/ConfirmationDialog.h:7]: @borisspektor, I didn't find why it worked earlier, but classes with UI prefix requires UIKit import.
